core/builtin.py

```diff
@@ -794,6 +794,24 @@ def Shopt(argv, exec_opts):
   return 0
 
 
+COMPGEN_SPEC = _Register('compgen')
+COMPGEN_SPEC.ShortFlag('-A', args.Str)
+
+
+def CompGen(argv, funcs):
+  arg, i = COMPGEN_SPEC.Parse(argv)
+  status = 0
+
+  if arg.A:
+    for func_name in sorted(funcs.iteritems()):
```

Comment:
The value of arg.A should be tested for `function` here.  The other actions can print a warning for "not implemented".